src/functions/module_lookup.ts

```diff
@@ -184,17 +187,22 @@ export const LOOKUP: AddFunctionDescription = {
     );
 
     if (nbCol > 1) {
-      assert(
-        () => index <= nbCol - 1,
-        _lt("[[FUNCTION_NAME]] evaluates to an out of range row value %s.", (index + 1).toString())
-      );
+      if (index >= nbCol) {
+        throw new NotAvailableError(
+          _lt("Did not find value '%s' in [[FUNCTION_NAME]] evaluation.", toString(searchKey))
+        );
+      }
       return resultRange[index][0] as FunctionReturnValue;
     }
 
-    assert(
-      () => index <= nbRow - 1,
-      _lt("[[FUNCTION_NAME]] evaluates to an out of range column value %s.", (index + 1).toString())
-    );
+    if (index >= nbRow) {
+      throw new NotAvailableError(
+        _lt(
+          "[[FUNCTION_NAME]] evaluates to an out of range column value %s.",
```

Comment:
maybe forgot to change the message here because the previous error message (```(index >= nbCol)```) changes